src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1012,7 +1015,7 @@ def get_create_and_insert_plan(child: SnowflakePlan, replace, error):
             )
 
         if mode == SaveMode.APPEND:
-            if self.session._table_exists(table_name):
+            if table_exists:
```

Comment:
if table_exists is None, this will fall int the else branch, right? shall we add an assertion here that table_exists can not be None when mode i append or truncate to be safe

Reply:
good point, added